File: android/app/src/main/java/vn/team/freechat/request/SendUserMessageRequest.java
package vn.team.freechat.request;

import com.tvd12.ezyfoxserver.client.entity.EzyData;
import com.tvd12.ezyfoxserver.client.factory.EzyEntityFactory;
import com.tvd12.ezyfoxserver.client.request.EzyRequest;

import vn.team.freechat.constant.Commands;

/**
 * Created by tavandung12 on 10/6/18.
 */

public class SendUserMessageRequest implements EzyRequest {

    private final long channelId;
    private final String message;

    public SendUserMessageRequest(long channelId, String message) {
        this.message = message;
        this.channelId = channelId;
    }

    @Override
    public EzyData serialize() {
        return EzyEntityFactory.newObjectBuilder()
                .append("channelId", channelId)
                .append("message", message)
                .build();
    }

    @Override
    public Object getCommand() {
        return Commands.CHAT_USER_MESSAGE;
    }
}

File: android/app/src/main/java/com/tvd12/freechat/request/SendUserMessageRequest.kt
package com.tvd12.freechat.request

import com.tvd12.ezyfoxserver.client.entity.EzyData
import com.tvd12.ezyfoxserver.client.factory.EzyEntityFactory
import com.tvd12.ezyfoxserver.client.request.EzyRequest
import com.tvd12.freechat.constant.Commands

/**
 * Created by tavandung12 on 10/6/18.
 */

class SendUserMessageRequest(
    private val channelId: Long,
    private val message: String
) : EzyRequest {

    override fun serialize() : EzyData {
        return EzyEntityFactory.newObjectBuilder()
                .append("channelId", channelId)
                .append("message", message)
                .build()
    }

    override fun getCommand() : String {
        return Commands.CHAT_USER_MESSAGE
    }
}